lib/socket.js

```diff
@@ -93,7 +93,6 @@ Socket.prototype.onPacket = function (packet) {
         break;
 
       case 'error':
-        this.transport.close();
```

Comment:
The reason of socket closing becomes "transport close" if we call here.
This method will be called on `clearTransport` anyway.